File: backend-spring-boot/src/test/java/no/vicx/backend/calculator/RemoveOldEntriesTaskTest.java
package no.vicx.backend.calculator;

import org.awaitility.Durations;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.bean.override.mockito.MockitoBean;
import org.springframework.test.context.bean.override.mockito.MockitoSpyBean;

import static org.awaitility.Awaitility.await;
import static org.mockito.Mockito.atLeast;
import static org.mockito.Mockito.verify;

@SpringBootTest
class RemoveOldEntriesTaskTest {

    @MockitoBean
    CalculatorService calculatorService;

    @MockitoSpyBean
    RemoveOldEntriesTask removeOldEntriesTask;

    @Test
    void removeOldEntriesTask_expectTimerToFire() {
        await().atMost(Durations.TEN_SECONDS).untilAsserted(() -> {
            verify(removeOldEntriesTask, atLeast(1)).removeOldCalculatorEntries();
            verify(calculatorService, atLeast(1)).deleteOldAnonymousCalculations();
        });
    }
}

File: backend-spring-boot/src/test/kotlin/no/vicx/backend/calculator/RemoveOldEntriesTaskTest.kt
package no.vicx.backend.calculator

import com.ninjasquad.springmockk.MockkBean
import io.kotest.assertions.nondeterministic.eventually
import io.kotest.core.spec.style.StringSpec
import io.mockk.verify
import org.springframework.boot.test.context.SpringBootTest
import kotlin.time.Duration.Companion.seconds

@SpringBootTest
class RemoveOldEntriesTaskTest(
    @MockkBean(relaxed = true) private val calculatorService: CalculatorService
) : StringSpec({

    "expect remove old entries task to fire" {
        eventually(10.seconds) {
            verify(atLeast = 1) { calculatorService.deleteOldAnonymousCalculations() }
        }
    }
})